### PR TITLE
Partially Fixes #4194: Added dark mode support to DrawerFragment, ConceptCardFragment and AudioFragment.

### LINKED ISSUE #4194: Add dark mode support to HomeActivity, DrawerFragment, RevisionCardActivity, ConceptCardFragment and AudioFragment
Add dark mode support to HomeActivity, DrawerFragment, RevisionCardActivity, ConceptCardFragment and AudioFragment

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by Akshatkamboj14**
## Explanation

Partially Fixes #4194 

**Drawer Fragment:** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/eb119ee3-033f-4757-8aee-7ea234d6f126/specs/

**Concept Card Fragment:** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/2ee531cf-c358-4a7d-b9dd-b772b51c3d3c/specs/

**Audio Fragment:** https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d67b268b-f6d9-434f-8897-24a2b13ccd84/specs/

**-> This PR is Broken into two pieces, First-one is to Add dark mode support to DrawerFragment, ConceptCardFragment, and AudioFragment. _The second is to Add dark mode support to HomeActivity and RevisionCardActivity (this is to be done by @bhaktideshmukh)_**

| **Default** | **Dark Mode** |
| ------------- | ------------- |
| image  | image  |
| image  | image  |
| image  | image)  |
| image | image  |
| image  | image  |

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Akshatkamboj14** [comment]
I have a doubt in the   `audio_fragment.xml` as there is textView that recently got added and the mocks are not up to date with the new text view, Please tell me if I have to add dark mode to that as well?

image

**Comment by rt4914** [comment]
@Akshatkamboj14 File a separate issue for the view/UI which does not have dark-mode implemented. That way it will be helpful in following cases:
1. We can ask designers to create mock and solve in separate PR
2. If we decide to solve this without mock than also its nice to track separately which issue got solved without mock and if needed to revert in future, it will be easy.

**Comment by Akshatkamboj14** [comment]
Okay sure @rt4914

**Comment by Akshatkamboj14** [comment]
Done @rt4914 
[https://github.com/oppia/oppia-android/issues/4561](url)

**Comment by Akshatkamboj14** [comment]
Hey @rt4914 PTAL, Thanks.

File: app/src/main/res/values/color_palette.xml [hunk 1]
```diff
@@ -91,4 +91,17 @@
...
+  <color name="color_palette_secondary_dark_background_color">@color/color_def_oppia_light_green</color>
+  <color name="color_palette_concept_card_toolbar_color">@color/color_def_oppia_brown_dark</color>
+  <color name="color_palette_menu_selected_text_color">@color/color_def_oppia_brown_dark</color>
+  <color name="color_palette_developer_option_menu_selected_color">@color/color_def_persian_blue</color>
+  <color name="color_palette_underline_view_color">@color/color_def_oppia_light_brown</color>
+  <color name="color_palette_menu_text_color">@color/color_def_black</color>
+  <color name="color_palette_secondary_container_background_color">@color/color_def_teal_blue</color>
+  <color name="color_palette_concept_status_bar_color">@color/color_def_dark_red</color>
 </resources>
+
```
Comment (rt4914) [inline app/src/main/res/values/color_palette.xml]: Remove this blank line

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Akshatkamboj14 Nice work. One minor change

File: app/src/main/res/values/color_palette.xml (hunk 1 above)
Comment (Akshatkamboj14) [reply app/src/main/res/values/color_palette.xml]: Done @rt4914

**Review by rt4914 — CHANGES_REQUESTED** [review]
@Akshatkamboj14 Merge with latest develop.

FYI you might need to remerge as a lot of conflicting PRs are getting merged daily.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**